crates/engine/tree/src/persistence.rs

```diff
@@ -204,12 +204,14 @@ pub enum PersistenceAction {
 pub struct PersistenceHandle {
     /// The channel used to communicate with the persistence task
     sender: Sender<PersistenceAction>,
+    /// The last persisted block number.
+    last_persisted_block_number: u64,
```

Comment:
hmm, this type is Clone

do we actually want this to be clone?

maybe we should track this value separately

> maybe better to store it in the tree?

I would prefer this